scilpy/io/streamlines.py

```diff
@@ -91,18 +91,54 @@ def save_from_voxel_space(streamlines, anat, ref_tracts, out_name):
     nib.streamlines.save(tracto, out_name, header=nib_object.header)
 
 
+def ichunk(sequence, n):
+    """ Yield successive n-sized chunks from sequence.
+
+    Parameters
+    ----------
+    sequence: numpy.ndarray
+        streamlines
+    n: int
+        amount of streamlines to load
+
+    Return
+    ------
+
+    chunck: list
+        subset of streamlines
+    """
+
+    sequence = iter(sequence)
+    chunk = list(islice(sequence, n))
+    while len(chunk) > 0:
+        yield chunk
+        chunk = list(islice(sequence, n))
+
+
 def load_tractogram_with_reference(parser, args, filepath,
-                                   bbox_check=True):
+                                   bbox_check=True, argName=None):
+
     _, ext = os.path.splitext(filepath)
     if ext == '.trk':
         sft = load_tractogram(filepath, 'same',
                               bbox_valid_check=bbox_check)
     elif ext in ['.tck', '.fib', '.vtk', '.dpy']:
-        if args.reference is None:
+        if argName:
+            if args.__getattribute__(argName + '_ref'):
```

Comment:
I would put that in a variable and re-use it (also a bit easier to understand if the variable name is clear)